app/models/metadata.rb

```diff
@@ -42,48 +42,26 @@ def #{association_name}
         build_#{association_name}
       end
 
-      def validating_ena_required_fields=(state)
```

Comment:
Were all these functions not required?

Reply:
These were always a bit of an abuse of Rails validation, using temporary instance variables to toggle certain validations on or off in particular contexts. A bug-fix in 5.2.4 prevented this from working, as essentially it was trying to detect if the record had changed before validation, and instance variables were ignored in this.

Fortunately Rails 4 added custom validation contexts, which provided exactly the functionality we needed, without the need for any of the old cruft. So while the code was still being used prior to the update, post update it was no-longer working, so has been replaced with the 'official' way of doing things.